### PR TITLE
added 'Internationalizing English Rust' project idea

### PR DESCRIPTION
**Pull request by feefladder**
See [Zulip chat](https://rust-lang.zulipchat.com/#narrow/stream/122651-general/topic/Internationalizing.20English.20Rust) for discussion. Any feedback is welcome!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -502,3 +503,69 @@ Small to medium (depends on the choice of implemented lints or schema extensions
...
+**Project size**
+
+Medium to large (depending on how far the stretches are taken).
+
+**Difficulty**
+
+Medium to hard.
+
+**Mentor**
+- Fee (Fairy) Fladder ([GitHub](https://github.com/feefladder/), [Zulip](HELP))
```
Comment (Nilstrieb) [inline README.md:552]: Isn't the mentor someone who's really familiar with the project?
I'm not sure whether you could mentor compiler or cargo changes :)

**Review by Nilstrieb — COMMENTED** [review]
This aims at making direct changes to Rust tooling, but there's also not been any RFC or anything like it to confirm that this is something we actually want today.
That seems like a bad idea to request people to work on things where we don't know whether we actually want them?

**Comment by Nilstrieb** [comment]
I like the motivation behind this, but something big as this needs a lot of design work first to address concerns and questions before anything is implemented. Examples of concerns I can think of right now: what about macros reading field names (like serde), how much would be expect popular crate authors to internationalize, what about rustdoc docs which seem more important to me etc. I think all of this can be addressed, but I don't think it can be in time before GSoC starts, and GSoC seems to mostly be about implementation work. Maybe this would be a good project in a few years when we have a clear design and approval from all relevant teams, but I think this year is too early to do this with the proper feature process.

**Comment by Kobzol** [comment]
I have basically the same comments/concerns as Nilstrieb.

If we want to add this to the idea list, it should be clearly stated that it would be an external project, implemented probably as an external compiler plugin (that's why I suggested to include it in the "ecosystem" category), that will **not** be modifying the compiler, because such a change wouldn't be accepted at the moment (not without an RFC anyway). As a second point, the mentor should have enough knowledge that they could do the project themselves (otherwise they won't be able to mentor the student properly). Do you think that you have enough experience to mentor someone to write a compiler plugin?

GSoC is indeed mostly for implementation work, this would need a lot of design up-front, but I suppose that some kind of demo could be implementable. We don't really have any strict requirements for including things in the idea list this year. For the current suggestions, the only requirement was that we have a mentor available who could actually mentor the student on the project.

Expanding on the requirements, there is probably also some notion of the Rust Project supporting the idea, if we put it on the idea list. Most of the current ideas came directly from the Rust Project members, and in that case it's clear that we support it. There are also some third-party ideas (the `cargo-semver-checks` ones), that are currently not affiliated with the Project, but since it is a developed tool that is very useful for the global ecosystem, we have also included it in the idea list (it also has a strong mentor). In this case, it is an idea that is not supported by the Project yet, and it also doesn't have any existing third-party project that it could extend (in fact, this project would need to be started). My inclination would thus be to perhaps bootstrap such a third-party solution first, and perhaps include it as a GSoC idea for some of the next years, once it has established itself and there are clear implementation goals that could improve it.

**Comment by Kobzol** [comment]
Ok, let's perhaps keep this idea for following years. Closing for now.

**Pull request closed without merging**